Content.Server/GameObjects/Components/Mobs/StunnableComponent.cs

```diff
@@ -0,0 +1,247 @@
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Timers;
+using Robust.Shared.Interfaces.Timing;
+using Robust.Shared.IoC;
+using Robust.Shared.Maths;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using Timer = Robust.Shared.Timers.Timer;
+
+namespace Content.Server.GameObjects.Components.Mobs
+{
+    [RegisterComponent]
+    public class StunnableComponent : Component, IActionBlocker, IAttackHand, IMoveSpeedModifier
+    {
+        public override string Name => "Stunnable";
+
+#pragma warning disable 649
+        [Dependency] private IEntitySystemManager _entitySystemManager;
+        [Dependency] private IGameTiming _gameTiming;
+#pragma warning restore 649
+
+        private TimeSpan? _lastStun;
+
+        [ViewVariables]
+        public TimeSpan? StunStart => _lastStun;
+
+        [ViewVariables]
+        public TimeSpan? StunEnd => _lastStun == null
+            ? (TimeSpan?) null
+            : _gameTiming.CurTime + TimeSpan.FromSeconds(_stunnedTimer + _knockdownTimer + _slowdownTimer);
+
+        private const int StunLevels = 8;
+
+        private bool _canHelp = true;
+        private float _stunCap = 20f;
+        private float _knockdownCap = 20f;
+        private float _slowdownCap = 20f;
+        private float _helpKnockdownRemove = 1f;
+        private float _helpInterval = 1f;
+
+        private float _stunnedTimer = 0f;
+        private float _knockdownTimer = 0f;
+        private float _slowdownTimer = 0f;
+
+        private float _walkModifierOverride = 0f;
+        private float _runModifierOverride = 0f;
+        private readonly string[] _texturesStunOverlay = new string[StunLevels];
+
+        [ViewVariables] public bool Stunned => _stunnedTimer > 0f;
+        [ViewVariables] public bool KnockedDown => _knockdownTimer > 0f;
+        [ViewVariables] public bool SlowedDown => _slowdownTimer > 0f;
+        [ViewVariables] public float StunCap => _stunCap;
+        [ViewVariables] public float KnockdownCap => _knockdownCap;
+        [ViewVariables] public float SlowdownCap => _slowdownCap;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _stunCap, "stunCap", 20f);
+            serializer.DataField(ref _knockdownCap, "knockdownCap", 20f);
+            serializer.DataField(ref _slowdownCap, "slowdownCap", 20f);
+            serializer.DataField(ref _helpInterval, "helpInterval", 1f);
+            serializer.DataField(ref _helpKnockdownRemove, "helpKnockdownRemove", 1f);
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            for (var i = 0; i < StunLevels; i++)
+            {
+                _texturesStunOverlay[i] = $"/Textures/UserInterface/Inventory/cooldown-{i}.png";
+            }
+        }
+
+        public void Stun(float seconds)
```

Comment:
Maybe add a simple boolean variable + check here so that people can temporarily become stun-immune (i.e. creatine)?

Reply:
Done.